fhir-profile/src/main/java/com/ibm/fhir/profile/ConstraintGenerator.java

```diff
@@ -304,13 +313,38 @@ private String generatePatternValueConstraint(ElementDefinition elementDefinitio
             CodeableConcept codeableConcept = pattern.as(CodeableConcept.class);
             Coding coding = codeableConcept.getCoding().get(0);
             String system = (coding.getSystem() != null) ? coding.getSystem().getValue() : null;
+
             sb.append(".where(coding.where(");
+
             if (system != null) {
                 sb.append("system = '").append(system).append("' and ");
             }
-            sb.append("code = '")
-                .append(coding.getCode().getValue())
-                .append("').exists()).exists()");
+
+            sb.append("code = '").append(coding.getCode().getValue()).append("').exists()).exists()");
+        } else if (pattern.is(Identifier.class)) {
+            Identifier _identifier = pattern.as(Identifier.class);
+            String system = _identifier.getSystem().getValue();
+
+            sb.append(".where(system = '").append(system).append("').count()");
+
+            Integer min = elementDefinition.getMin().getValue();
+            String max = elementDefinition.getMax().getValue();
+
+            if ("*".equals(max)) {
+                sb.append(" >= ").append(min);
+            } else if ("1".equals(max)) {
+                if (min == 0) {
+                    sb.append(" <= 1");
+                } else {
+                    sb.append(" = 1");
+                }
+            } else {
+                sb.append(" >= ").append(min).append(" and ").append(identifier).append(".where(system = '").append(system).append("').count() <= ").append(max);
+            }
+
+            if (!node.children.isEmpty()) {
+                sb.append(" and (").append(identifier).append(".where(system = '").append(system).append("').exists()").append(" implies (").append(identifier).append(".where(system = '").append(system).append("' and ").append(generate(node.children)).append(")))");
```

Comment:
any way to format this to make it a little simpler to see whats going on?

Reply:
or maybe just a comment to indicate what this and clause is doing (at a high level)?

Reply:
The way the constraint is generated, there isn't much in the way of indentation that will do it justice. Looking at the actual generated constraint is more helpful.